vendor/github.com/ethereum/go-ethereum/p2p/peer.go

```diff
@@ -248,7 +264,24 @@ func (p *Peer) pingLoop() {
 	}
 }
 
-func (p *Peer) readLoop(errc chan<- error) {
+func (p *Peer) heartbeatLoop(reads <-chan struct{}) {
+	defer p.wg.Done()
+	hb := time.NewTimer(2 * pingInterval)
```

Comment:
`const heartbeatInterval`?

Reply:
but frankly, that looks more like a watchdog timer.

Reply:
thanks, watchdog sounds more accurate